IndexCoop/index-app
# fix: approvals

⚠️ DO NOT MERGE WITHOUT REVERTING THE LAST COMMIT ⚠️ 

## **Summary of Changes**

* Fixes approval issues where the approval state would not be recognized (after success), not updating the trade button

## Docs
* https://1.x.wagmi.sh/react/hooks/useContractRead

## Testing

* [e0dec2d](https://github.com/IndexCoop/index-app/pull/1063/commits/e0dec2d3ae528637f33984992e24d1d235ab9a2b) is necessary for testing to work but will have to be reverted before merging! ⚠️ 
* Start hardhat node: `npm run hardhat`
* Fund one of the testing accounts (address specified in .env.local): `npm run hardhat:fund`
* Connect to `localhost`
* Use USDC as currency e.g. USDC-ETH2X
* Input amount e.g. 10 USDC
* Test approving (and state changes of button) 🧪 🔬 

In case you restart the hardhat node and receive nonce issues with the test account, follow these instructions to reset the account data in metamask: https://support.metamask.io/hc/en-us/articles/360015488891-How-to-clear-your-account-activity-reset-account. Make sure that the test account is selected!

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.
